vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethod.java

```diff
@@ -79,6 +85,11 @@ public Void call() throws Exception {
 
 				log.debug("Invoking {}", reflectionMethod);
 				Object instance = event.getControllerInstance();
+				
+				Class<?> declaringClass = reflectionMethod.getDeclaringClass();
+				if (!declaringClass.isAssignableFrom(instance.getClass())) {
+					instance = container.instanceFor(declaringClass);
```

Comment:
Yes. The declaringClass is the original/first controller. In the case of issue #977, it'd be the IndexController.